src/Microsoft.OpenApi/Validations/Visitors/OpenApiVisitorSet.cs

```diff
@@ -14,39 +16,23 @@ namespace Microsoft.OpenApi.Validations.Visitors
     /// </summary>
     internal static class OpenApiVisitorSet
     {
-        private static IDictionary<Type, IVisitor> _visitorCache = new Dictionary<Type, IVisitor>
+        private static IDictionary<Type, IVisitor> _visitors;
+
+        /// <summary>
+        /// Gets the visitors
+        /// </summary>
+        public static IDictionary<Type, IVisitor> Visitors
         {
-            { typeof(OpenApiCallback), new CallbackVisitor() },
-            { typeof(OpenApiComponents), new ComponentsVisitor() },
-            { typeof(OpenApiContact), new ContactVisitor() },
-            { typeof(OpenApiDiscriminator), new DiscriminatorVisitor() },
-            { typeof(OpenApiDocument), new DocumentVisitor() },
-            { typeof(OpenApiEncoding), new EncodingVisitor() },
-            { typeof(OpenApiExample), new ExampleVisitor() },
-            { typeof(OpenApiExternalDocs), new ExternalDocsVisitor() },
-            { typeof(OpenApiHeader), new HeaderVisitor() },
-            { typeof(OpenApiInfo), new InfoVisitor() },
-            { typeof(OpenApiLicense), new LicenseVisitor() },
-            { typeof(OpenApiLink), new LinkVisitor() },
-            { typeof(OpenApiMediaType), new MediaTypeVisitor() },
-            { typeof(OpenApiOAuthFlows), new OAuthFlowsVisitor() },
-            { typeof(OpenApiOAuthFlow), new OAuthFlowVisitor() },
-            { typeof(OpenApiOperation), new OperationVisitor() },
-            { typeof(OpenApiParameter), new ParameterVisitor() },
-            { typeof(OpenApiPathItem), new PathItemVisitor() },
-            { typeof(OpenApiPaths), new PathsVisitor() },
-            { typeof(OpenApiRequestBody), new RequestBodyVisitor() },
-            { typeof(OpenApiResponses), new ResponsesVisitor() },
-            { typeof(OpenApiResponse), new ResponseVisitor() },
-            { typeof(OpenApiSchema), new SchemaVisitor() },
-            { typeof(OpenApiSecurityRequirement), new SecurityRequirementVisitor() },
-            { typeof(OpenApiSecurityScheme), new SecuritySchemeVisitor() },
-            { typeof(OpenApiServerVariable), new ServerVariableVisitor() },
-            { typeof(OpenApiServer), new ServerVisitor() },
-            { typeof(OpenApiTag), new TagVisitor() },
-            { typeof(OpenApiXml), new XmlVisitor() }
-        };
+            get
+            {
+                if (_visitors == null)
+                {
+                    _visitors = new Lazy<IDictionary<Type, IVisitor>>(() => BuildVisitorSet(), isThreadSafe: false).Value;
```

Comment:
Maybe Lazy is unnecessary?

Reply:
I would say the underlying operation is not that time consuming/complex, so I think it's unnecessary. But I don't really mind using it if you prefer.

---
In reply to: [157624690](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157624690) [](ancestors = 157624690)

Reply:
Ok.

---
In reply to: [157634816](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157634816) [](ancestors = 157634816,157624690)